modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/clause/SQLSelectClause.java

```diff
@@ -192,46 +232,75 @@ private String buildLimit(final SQLContext context) {
         return selectPredicate;
     }
 
-    private String buildFrom(final SQLContext context) {
+    private String buildFrom(final SQLContext context) throws EdmException {
         List<String> tables = new ArrayList<>();
         Iterator<String> it = query.getTablesAliasesForEntitiesInQuery();
         while (it.hasNext()) {
             String tableAlias = it.next();
-            EdmStructuralType target = query.getEntityInQueryForAlias(tableAlias);
-            if (isSelectTarget(target)) {
-                tables.add(query.getSQLTableName(target) + " AS " + tableAlias);
+            EdmStructuralType type = query.getEntityInQueryForAlias(tableAlias);
+            if (isSelectTarget(type)) {
+                if (this.parameterMapping.isEmpty()) {
+                    tables.add(query.getSQLTableName(target) + " AS " + tableAlias);
+                } else {
+                    tables.add(query.getSQLTableName(target) + buildTargetParameters(context) + " AS " + tableAlias);
+                }
             }
         }
         return SQLUtils.csv(tables);
     }
 
+    private String buildTargetParameters(final SQLContext context) throws EdmException {
+        String targetDataStructureType = this.query.getSQLTableDataStructureType(this.target);
+        StringBuilder parameters = new StringBuilder();
+        Iterator<Integer> pi = parameterMapping.keySet().iterator();
+
+        while (pi.hasNext()) {
+            Integer parameter = pi.next();
+            EdmStructuralType type = getTargetTypeFromParameterMapping(parameter);
+            EdmProperty prop = getPropertyFromParameterMapping(parameter);
+            EdmSimpleType edmSimpleType = (EdmSimpleType) prop.getType();
+            String propertyName = getPropertyNameFromParameterMapping(parameter);
+            Object parameterValue = getKeyPredicateValueByPropertyName(propertyName, this.keyPredicates);
+            parameterValue = evaluateDateTimeExpressions(parameterValue, edmSimpleType);
+            EdmTableBinding.ColumnInfo info = this.query.getSQLTableColumnInfo(type, prop);
+            this.statementParams.add(new SQLStatementParam(parameterValue, prop, info));
+
+            if (context.getDatabaseProduct().equals(DatabaseProduct.HANA) && targetDataStructureType.equals(ISqlKeywords.METADATA_CALC_VIEW)) {
+                parameters.append("placeholder.\"$$" + propertyName + "$$\"" + " => " + " ? ");
+            }
+
+            if (pi.hasNext()) {
+                parameters.append(",");
+            }
+        }
+
+        return "(" + parameters + ")";
+    }
+
```

Comment:
What if DB is not HANA? This will return () can we move the check from line 268 up so that it skips the whole method execution if DB type is not HANA?